### PR TITLE
Added make i18n to volto

### LINKED ISSUE #6263: Add a make command for `yarn i18n`, such as `make i18n`
Should we add a make command for `yarn i18n`, such as `make i18n`? We've been trying to unify all commands in the Makefiles.

Here's the output from the command.

```console
$ yarn i18n    
yarn run v1.22.19
$ pnpm --filter @plone/volto run i18n

> @plone/volto@18.0.0-alpha.42 i18n /Users/stevepiercy/projects/Plone/documentation/submodules/volto/packages/volto
> rm -rf build/messages && NODE_ENV=production i18n

Extracting messages from source files...
Synchronizing messages to pot file...
Synchronizing messages to po files...
Generating the language JSON files...
Merging @plone/volto-slate locales for de
Merging @plone/volto-slate locales for en
done!
✨  Done in 6.71s.

```

Here's the recently updated documentation:

https://6.docs.plone.org/i18n-l10n/contributing-translations.html#translate-volto

_Originally posted by @stevepiercy in https://github.com/plone/documentation/issues/1691#issuecomment-2300039202_

### PR DESCRIPTION
**Pull request by Faakhir30**
Fixes #6263 
Added make i18n to Volto's makefile.

sample output:

```

f@f:~/workspaces/oss/plone/plone-volto$ make i18n
make[1]: Entering directory '/home/f/workspaces/oss/plone/plone-volto/packages/volto'

> @plone/volto@18.0.0-alpha.42 i18n /home/f/workspaces/oss/plone/plone-volto/packages/volto
> rm -rf build/messages && NODE_ENV=production i18n

Extracting messages from source files...
Synchronizing messages to pot file...
Synchronizing messages to po files...
Generating the language JSON files...
Merging @plone/volto-slate locales for de
Merging @plone/volto-slate locales for en
done!
make[1]: Leaving directory '/home/f/workspaces/oss/plone/plone-volto/packages/volto'

```

Volto's default package-manager in package.json is pnpm, so using `pnpm i18n`

Will update in plone/documentation repo also once this is confirmed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,6 +75,10 @@ install: ## Set up development environment
 	node packages/scripts/vscodesettings.js
 	make build-deps
 
+.PHONY: i18n
+i18n: ## Converts your po files into json to translate volto frontend
+	$(MAKE) -C "./packages/volto/" i18n
+
```
Comment (stevepiercy) [inline Makefile:81]: This belongs below the `.PHONY: build-deps` target, under the `## Build` section.

File: packages/volto/CHANGELOG.md
Comment (stevepiercy) [inline packages/volto/CHANGELOG.md:1]: Do not edit this file. See https://6.docs.plone.org/contributing/index.html#contributing-change-log-label

Comment (stevepiercy) [reply packages/volto/CHANGELOG.md:1]: Also do not include trim of whitespace in a pull request, as that just adds noise to what actually needs review. Instead create a separate PR for trims.

File: packages/volto/Makefile [hunk 2]
```diff
@@ -66,6 +66,10 @@ install: ## Set up development environment
 	pnpm i
 	make build-deps
 
+.PHONY: i18n
+i18n: ## Extract and compile translations
+	pnpm i18n
```
Comment (stevepiercy) [inline packages/volto/Makefile:71]: This belongs below the `.PHONY: build-deps` target, under the `## Build` section.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
I confirm that this works.

I have a few suggestions, and once you make them, I'm happy to merge this PR. Thank you!

File: packages/volto/news/6274.internal [hunk 3]
```diff
@@ -0,0 +1 @@
+Add `make i18n` command to volto. @Faakhir30
```
Comment (stevepiercy) [inline packages/volto/news/6274.internal:1]:
```suggestion
Add `make i18n` command. @Faakhir30
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Review by stevepiercy — APPROVED** [review]
Looks good. Let's merge it when all checks pass.

**Comment by stevepiercy** [comment]
@Faakhir30 would you also please submit a PR in `plone/documentation` at https://6.docs.plone.org/i18n-l10n/contributing-translations.html#translate-volto? Thank you!

**Comment by Faakhir30** [comment]
> @Faakhir30 would you also please submit a PR in `plone/documentation` at https://6.docs.plone.org/i18n-l10n/contributing-translations.html#translate-volto? Thank you!

yeah sure, it'll be done by today / tomorrow, thanks

**Pull request merged**